### PR TITLE
add Third Party Programs file

### PR DESCRIPTION
**Pull request by bratpiorka**
add Third Party Programs file

### CODE REVIEW

File: third-party-programs.txt [hunk 1]
```diff
@@ -0,0 +1,570 @@
+Unified Memory Framework (UMF) Third Party Programs File
```
Comment (lukaszstolarczuk) [inline third-party-programs.txt:1]: should we install this file (along with LICENSE)...?

Comment (bratpiorka) [reply third-party-programs.txt:1]: done

File: test/test_installation.py [hunk 2]
```diff
@@ -146,6 +146,7 @@ def _create_match_list(self) -> List[str]:
         examples.insert(0, "share/doc/umf/examples")
         share.extend(examples)
         share.append("share/doc/umf/LICENSE.TXT")
+        share.append("share/doc/umf/third-party-programs.TXT")
```
Comment (lukaszstolarczuk) [inline test/test_installation.py:149]: `txt` lowercase, I guess

File: third-party-programs.txt [hunk 3]
```diff
@@ -0,0 +1,570 @@
...
+    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+    IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+    AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+    FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+    DEALINGS IN THE SOFTWARE.
+_______________________________________________________________________________
+
+7. The Unified Runtime Project is under the Apache License v2.0 with LLVM
```
[the remark this answers is not in the document]
Comment (bratpiorka) [reply third-party-programs.txt:347]: Yes, some code (disjoint Pool) was copied from UR

File: test/test_installation.py (hunk 2 above)
Comment (bratpiorka) [reply test/test_installation.py:149]: thanks, done

File: third-party-programs.txt [hunk 4]
```diff
@@ -0,0 +1,570 @@
...
+    IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+    AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+    FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+    DEALINGS IN THE SOFTWARE.
+
+_______________________________________________________________________________
+
+2.  Portable Hardware Locality (hwloc):
```
Comment (lplewa) [inline third-party-programs.txt:36]: Do we copy, or distribute hwloc? if not we don't need it here

File: third-party-programs.txt [hunk 5]
```diff
@@ -0,0 +1,570 @@
...
+    IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT
+    NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+    DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+    THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+    (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
+    THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+_______________________________________________________________________________
+
+3. jemalloc
```
Comment (lplewa) [inline third-party-programs.txt:87]: Do we copy, or distribute jemalloc? if not we don't need it here

File: third-party-programs.txt [hunk 6]
```diff
@@ -0,0 +1,570 @@
...
+    DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+    (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+    SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+    OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+    DAMAGE.
+_______________________________________________________________________________
+
+4. TBB malloc
```
Comment (lplewa) [inline third-party-programs.txt:115]: dito

File: third-party-programs.txt (hunk 4 above)
Comment (bratpiorka) [reply third-party-programs.txt:36]: there is an option where hwloc is statically linked with UMF (UMF_LINK_HWLOC_STATICALLY), so yes this is needed

File: third-party-programs.txt (hunk 5 above)
Comment (bratpiorka) [reply third-party-programs.txt:87]: if we link with jemalloc statically, then yes

File: third-party-programs.txt (hunk 6 above)
Comment (bratpiorka) [reply third-party-programs.txt:115]: agree - we do not link or use TBB headers anywhere - removed

**Review by lukaszstolarczuk — APPROVED** [review]

**Review by lplewa — APPROVED** [review]

**Review by lplewa — APPROVED** [review]

**Pull request merged**